contributors/design-proposals/encryption.md

```diff
@@ -0,0 +1,403 @@
+# Encryption
+
+## Abstract
+
+The scope of this proposal is to ensure that resources can be encrypted at the
+datastore layer with sufficient metadata support to enable integration with
+multiple encryption providers and key rotation. Encryption will be optional for
+any resource, but will be used by default for the Secret resource. Secrets are
+already protected in transit via TLS.
+
+Full disk encryption of the volumes storing etcd data is already expected as
+standard security hygiene.  Adding the proposed encryption inside the database
+defends against malicious parties gaining access to:
+
+- The etcd API
+- etcd backups; or
+- A running etcd instance without access to memory of the etcd process.
```

Comment:
the keys are encrypted before hitting etcd, how does it change with or without memory access?